netlify/edge-functions/github-auth.ts

```diff
@@ -69,8 +69,8 @@ async function getUser(token: string) {
     throw new Error("Error getting user's emails")
   }
 
-  const emails = (await emailResponse.json()) as Array<{ email: string; primary: boolean }>
-  const primaryEmail = emails.find((email) => email.primary)
+  const emails = (await emailResponse.json()) as Array<{ email: string; primary: boolean; visibility: string }>
+  const primaryEmail = emails.find((email) => email.primary && email.visibility === "public")
 
   if (!primaryEmail) {
     throw new Error("No primary email found")
```

Comment:
Should we update the error message to say "No public primary email found" and include a link to where they can manage their emails?

Reply:
It would be nice to let people know where they can manage their privacy settings, but doing it in the developer console won't be very helpful for regular users.